### PR TITLE
Notes parsing scans for reserved words

### PR DESCRIPTION
**Pull request by newduke**
Also forbids suit abbreviation to be a reserved word.

### CODE REVIEW

File: client/src/game/reducers/constants.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "e5cm",
+  "tcm",
+  "tccm",
+  "sdcm",
+  "sbpcm",
+  "ocm",
+  "tocm",
+  "utfcm",
+  "utbcm",
+];
```
Comment (Zamiell) [inline client/src/game/reducers/constants.ts:15]: lets also add utf for unnecessary trash finesse

Comment (Zamiell) [reply client/src/game/reducers/constants.ts:15]: and utdcm for unnecessary trash discharge chop move

File: client/src/game/reducers/constants.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export const KNOWN_TRASH_NOTES = ["kt", "trash", "stale", "bad"];
+export const FINESSED_NOTES = ["f", "hf", "pf", "gd"];
+export const NEEDS_FIX_NOTES = ["fix", "fixme", "needs fix"];
+export const BLANK_NOTES = ["blank"];
+export const UNCLUED_NOTES = ["unclued"];
+export const CLUED_NOTES = ["clued"];
+export const ALL_RESERVED_NOTES = KNOWN_TRASH_NOTES.concat(
```
Comment (Zamiell) [inline client/src/game/reducers/constants.ts:22]: write it like this:
[].concat(
  KNOWN_TRASH_NOTES,

Comment (newduke) [reply client/src/game/reducers/constants.ts:22]: Ugh, the typescript pedantry.

File: client/src/game/reducers/constants.ts [hunk 3]
```diff
@@ -14,12 +14,13 @@ export const CHOP_MOVED_NOTES = [
   "utbcm",
 ];
 export const KNOWN_TRASH_NOTES = ["kt", "trash", "stale", "bad"];
-export const FINESSED_NOTES = ["f", "hf", "pf", "gd"];
+export const FINESSED_NOTES = ["f", "hf", "pf", "gd", "utf"];
```
Comment (Zamiell) [inline client/src/game/reducers/constants.ts:17]: and utdcm for unnecessary trash discharge chop move

File: client/src/game/reducers/constants.ts [hunk 4]
```diff
@@ -14,12 +14,13 @@ export const CHOP_MOVED_NOTES = [
...
 export const NEEDS_FIX_NOTES = ["fix", "fixme", "needs fix"];
 export const BLANK_NOTES = ["blank"];
 export const UNCLUED_NOTES = ["unclued"];
 export const CLUED_NOTES = ["clued"];
-export const ALL_RESERVED_NOTES = KNOWN_TRASH_NOTES.concat(
+export const ALL_RESERVED_NOTES = (<string[]>[]).concat(
```
Comment (Zamiell) [inline client/src/game/reducers/constants.ts:22]: write it like this:
```
export const ALL_RESERVED_NOTES: string[] = [].concat(
```
i think that should work

Comment (newduke) [reply client/src/game/reducers/constants.ts:22]: nope.

Comment (Zamiell) [reply client/src/game/reducers/constants.ts:23]: damn

**Comment by Zamiell** [comment]
thank you good sir

**Pull request merged**